### DISCUSSION TITLE
Notice of Compromised Fairness in Quiz Round Game and Proposed Solutions for Participants to Vote On

### DISCUSSION THREAD

**Discussion by taramakage** [game-of-nfts/gon-evidence#671] [Polls]
We are very sorry to inform you that the fairness of this round of the Quiz Round game has been compromised due to a loophole in the rule design, and we deeply regret this.

Originally, our intention was to distribute 100 NFTs through airdrop and let participants solve the problems and complete the transfer. We estimated that these tasks would be completed equally among the participants. However, unexpectedly, a large number of NFTs were concentrated in a few accounts, which has compromised the fairness of the game.

Here, we propose the following solutions and invite participants to vote to decide:

1. Restart Round 5 and Limit each person to complete no more than 1 or 2 tasks to avoid the situation where a large number of NFTs are concentrated in a few accounts.
2. Adjust the score of the Quiz task from 10 points to 2 points to maintain the fairness of task allocation.
3. Cancel this round of the game to avoid further affecting the fairness of the game.

We hope that participants can express their opinions and ideas in this round of the game and vote to choose the measures they support. We will take appropriate measures as soon as possible after the voting results are released to ensure the fairness and justice of the game.

We apologize again for the inconvenience caused by this situation, and we will do our best to improve the game rules and design to ensure the fairness and justice of the the game. Thank you for your understanding and support.

~~**The deadline for voting is March 24th at 18:00 UTC.**~~ 

**While we understand that many voted to cancel the round in the discussion, we also recognize the possibility that the voting process might have been influenced by the uneven distribution of points among the teams.**

**Comment by gjermundgaraba** [comment]
Reposting and adding some comments from Discord: I realize I've caused some issues there. I didn't anticipate this what I did was considered a loop hole. I do indeed have 41 of the quiz questions. 

What I did was simply to write a small program that showed me the questions and let me answer them and see if the response was correct or not. If they answer was correct I imported the mnemonic and transferred the NFTs according to the flow. But it was not a hack or anything, it just made it slightly more convenient to answer questions. And I did spend a lot of time making this. And answering and transferring also took a lot of time.

I don't see it as a loophole, but I also understand that it sucks that so few people got to answer the question. But I felt like it was a competition and therefore become somewhat competitive. I did spend many hours doing all of this, so not sure exactly how unfair it really is though.

In the end my intention was not to frustrate or anything, but I feel like I played by the rules and put a lot of effort into doing well - so it would suck if this round was simply canceled and I was left with nothing after the effort I put into it.

**Comment by nodeiistt** [comment]
Discord repost: 
The Stage 5 Algorithm is completely wrong.
I'm not talking about bots.


What I'm talking about is that there are hundreds of testnet participants.
and that's more than all the points hundreds of people earned in 1 month
Earned in less than 1 hour with Stage 5 missions.

So the algorithm is wrong... Don't these guys say "Why are we struggling for 1 month"? I think they say. and they are right...

Stage 5 trashed all other stages.


Reasons:
- A very high score.
- There is no limit to doing the task over and over again.
- It is also much easier than many other flows.
- It has the potential to easily abuse the testnet. and only 2 wallets solved more than 60 of about 100 quizzes.
What I want to say is that this is not normal.

Also, there is no technique. It's purely a fun stage.
There are no different flows. all the same f04.
After all, this is not a quizzes activity. testnet activity. Maybe a separate event can be organized for this.

If the aim is to compete in technical issues, as I mentioned above,
Instead of 100 quizzes, have 100 different flows, and everyone can test these flows.
anyone can do all 100 of 100 flows. 10 points per flow 
I found the answers to more than 40 questions one by one and tried it manually without using any bots.

I deleted and added wallets for all of them one by one.
I believe I put more effort into this than writing bots. but the result is disappointing 
And it's not just me, but probably dozens of others.

**Comment by Exclusive048** [comment]
I don't think it's worth canceling this quiz.
For example, I did all my 15 questions myself. I just looked at the address in the browser, and if there was nft I quickly tried to answer the question and follow the next steps.

If you consider this an incorrect task, then the task that was performed with self-relay can also be considered a loophole from which two challenges were made.

I am grateful for the help @gjermund | EmpowerChain , but not everyone could count on it and not everyone could do it personally.


Just wanted to let you know. Who prevented you from being faster and more agile. This is the essence of the quiz.

In particular, @gjermundgaraba and I completed 56 tasks out of 100 possible. If the participants were more active, they could compete with each other. 

Key:
If we go down the path of canceling / restarting jobs, we will never come to a common value.

You can't please everyone. Everyone can't do well. There will always be dissatisfied.

The only thing that is possible is to slightly lower the score for the question. But it will not be fair to those who answered the questions and wasted their time.
If you want to reduce the score, then I ask you to restart the task B5-B7. Since I was having problems with the Internet and could not quickly respond to an error with 3 NFTs, I increased the transfer time by 300 blocks.

Also. I would like to say that the questions were quite important. Basis, standards, interchain etc.

If you consider these questions not important, then I can say the same about the rest of the tasks.
They were the same type, you all had time to prepare. Everyone in the chat could HELP you and even WRITE A COMMAND so that you complete the task.

WHERE IS THE TECHNIQUE?

Sorry for the large font. I'm a little outraged.

**Comment by gjermundgaraba** [comment]
It is a waste of time to have a vote on this. I can tell you the outcome every time: cancel. Why? It's obvious: almost everyone has a strong incentive to cancel since it wasn't a participation medal. If it wasn't meant to be a competitive game, it should have been designed just like the other tasks: everyone gets a point. However, the rules set a different tone: "The total number of NFTs is only 100, and each player can solve as many questions as possible to get points." I spent time and energy doing just that (without cheating or using any loopholes), so removing that seems unfair as well, no?

What I don't like about this is mostly the spirit of a game is kinda ruined by this. If you can change the outcome of the game just because many participants don't like it, what's the point? It doesn't exactly breed the competitive spirit of a game 🙂 But maybe that is not desired either.
(If there was an exploit or some loophole had been used here, it would've been one thing, but this wasn't that)

**Reply by Exclusive048** [reply]
I fully support this, but don't expect the task to be cancelled.
Everyone, except for those who received points for the quiz, were happy to cancel this task.
Let me give you an example from sports. The top 1 - A and top 100 - B teams are playing. With a 95% probability, team A will win and score more points, but note that team B will not be indignant that the players are faster / more accurate / stronger, it doesn’t matter. This is the competition, otherwise what is the point.

**Reply by giansalex** [reply]
this round was badly planned

**Reply by Exclusive048** [reply]
Even though the time was not exactly right, for most of the participants it was prime time, but they were in no hurry to complete the quiz. And now they want to cancel the results because they misunderstood the rules. That's bullshit.
I'm waiting for complaints about the winners of the transfer speed tasks: B3, B4, B8. This is also a kind of loophole. After all, some participants are younger and they have fast hands, or they are more experienced or something else.😀

**Comment by taramakage** [comment]
We want to express our gratitude for your active participation and valuable feedback during the quiz competition. We have been closely following the discussions and concerns raised about Round 5, where most points were secured by a few teams. While we understand that many voted to cancel the round in the discussion, we also recognize the possibility that the voting process might have been influenced by the uneven distribution of points among the teams. After careful consideration, Round 5 will remain valid and the points earned according to the rules will be counted. Considered the following points in the decision: 
- Flow Complexity: The length of a flow is not necessarily proportional to its difficulty. In fact, after Stage 1 and 2, the type of flow becomes less important, as the underlying principles of ports and channels is the same, and the flow itself is just a change in form.
- Effort & Workload: Solving quizzes, decoding mnemonics, and executing transfers require a higher level of effort than regular transfers.
- Competition Preparation: Rules and points were made available approximately 1 day prior, allowing to review the rules, prepare strategies, or raise questions. 
- Competitive Nature: This is designed to competitive and fast-paced, and speed is crucial in Stage 3. 

**Challenge tasks** and **Awards** are still open for participation. Remember to name your issue in the award claim form before the deadline if you submitted bug reports and wish to apply for awards. We value your cooperation and understanding.

**Reply by giansalex** [reply]
> Effort & Workload: Solving quizzes, decoding mnemonics, and executing transfers require a higher level of effort than regular transfers.

but why this round is so highly valued, is "Solving quizzes, decoding mnemonics" the objective of this competition, should it not be related to the operation of ICS721?

**Reply by taitruong** [reply]
> We want to express our gratitude for your active participation and valuable feedback during the quiz competition. We have been closely following the discussions and concerns raised about Round 5, where most points were secured by a few teams. While we understand that many voted to cancel the round in the discussion, we also recognize the possibility that the voting process might have been influenced by the uneven distribution of points among the teams. After careful consideration, Round 5 will remain valid and the points earned according to the rules will be counted. Considered the following points in the decision:
> 
> * Flow Complexity: The length of a flow is not necessarily proportional to its difficulty. In fact, after Stage 1 and 2, the type of flow becomes less important, as the underlying principles of ports and channels is the same, and the flow itself is just a change in form.
> * Effort & Workload: Solving quizzes, decoding mnemonics, and executing transfers require a higher level of effort than regular transfers.
> * Competition Preparation: Rules and points were made available approximately 1 day prior, allowing to review the rules, prepare strategies, or raise questions.
> * Competitive Nature: This is designed to competitive and fast-paced, and speed is crucial in Stage 3.
> 
> **Challenge tasks** and **Awards** are still open for participation. Remember to name your issue in the award claim form before the deadline if you submitted bug reports and wish to apply for awards. We value your cooperation and understanding.

Ok fine, but why do you offer and have put a vote on this, then overruling it after all?

**Comment by taitruong** [comment]
As stated by @taramakage:

```
Originally, our intention was to distribute 100 NFTs through airdrop and let participants solve the problems and complete the transfer. **We estimated that these tasks would be completed equally among the participants.** However, unexpectedly, a large number of NFTs were concentrated in a few accounts, which has compromised the fairness of the game.
```

All 100 quizzies have been solved, instead of being equally completed by participants. Also 2nd proposal:

```
2. Adjust the score of the Quiz task from 10 points to 2 points to maintain the fairness of task allocation.
```

What is the original intend 10 points per quiz? So if someone solves 50 quizzies, participant would get 500 points? This doesn't sound balanced to all other challenges, claims and bug awards.

**Comment by marcotradenet** [comment]
In my opinion, the rules are not very clear:  
```
The participants must then use the mnemonics to generate these preset addresses AND TRANSFER THE NFT ACCORDING TO THE FLOW.
```
I had understood that one should transfer the nft immediately through flow with wallet from mnemonics and then to finish transfer to the participation wallet. 
Instead, it seems that everyone who solved the quizzes first transferred the nft to a wallet and then calmly did the flow.

For me the competition was also fine (whoever can be more capable deserves to have more points), but maybe the rules should be discussed first and if something is not clear it should be explained better.

Personally, I wasted time setting up the flow before transfer the nft from original wallet, so I couldn't get a single one.

**Reply by Exclusive048** [reply]
You had a whole day to set up the flow, 24 hours to set up the flow. What prevented you from doing this?

The decision of the administration has already been announced. I see no reason to continue discussions.

**Comment by giansalex** [comment]
@taramakage it will be verified that the NFTs were not obtained using the exploit #346? then they would not need to answer questions or decode mnemonics
